pywinauto/base_application.py

```diff
@@ -242,22 +291,189 @@ def __resolve_control(self, criteria, timeout=None, retry_interval=None):
             ctrl = wait_until_passes(
                 timeout,
                 retry_interval,
-                self.__get_ctrl,
+                self.__find_base,
                 (findwindows.ElementNotFoundError,
                  findbestmatch.MatchError,
                  controls.InvalidWindowHandle,
                  controls.InvalidElement),
-                criteria)
+                self.criteria,
+                timeout,
+                retry_interval,
+            )
+        except TimeoutError as e:
+            raise e.original_exception
+
+        return ctrl
 
+    def find_all(self, timeout=None, retry_interval=None):
+        """
+        Find all controls using criteria. The returned controls match conditions from criteria[-1].
+        Parent controls are assumed to exist in a single instance. Otherwise it will result in an AmbiguousError.
+
+        * **criteria** - a list with dictionaries
+
+             1st element is search criteria for the dialog
+
+             other elements are search criteria for a control of the dialog
+
+        * **timeout** -  maximum length of time to try to find the controls (default 5)
+        * **retry_interval** - how long to wait between each retry (default .2)
```

Comment:
Current default is `'window_find_retry': .09,` in timings.py. I know this is old issue, but please fix it.